### PR TITLE
change E to int64 to fit bigger exponents we see in the wild

### PR DESCRIPTION
**Pull request by maxtaco**
- fork RSA from Go 1.5.3 :(

### CODE REVIEW

**Comment by maxtaco** [comment]
Confirmed this fixes the problem with `keybase id cr0bar`

**Comment by oconnor663** [comment]
Oh dear. Is there a way to see a diff of what we changed, apart from the copies?

**Comment by cr0bar** [comment]
@oconnor663 I can see the changes here:

https://github.com/keybase/go-crypto/pull/8/files

Change to import, 1 if statement and the type change to int64

Both public key and public key v3 files changed.

**Comment by cr0bar** [comment]
Ignore my comment, realise what you mean now.

**Comment by maxtaco** [comment]
Our options are limited since the library we need to fix is in go core

**Comment by maxtaco** [comment]
the only change in the RSA library is from `int` to `int64` and a few fixed
#-of bits computations

File: openpgp/packet/public_key_v3.go [hunk 1]
```diff
@@ -100,14 +100,14 @@ func (pk *PublicKeyV3) parseRSA(r io.Reader) (err error) {
 	if len(pk.n.bytes) < 8 {
 		return errors.StructuralError("v3 public key modulus is too short")
 	}
-	if len(pk.e.bytes) > 3 {
+	if len(pk.e.bytes) > 7 {
```
Comment (oconnor663) [inline openpgp/packet/public_key_v3.go:103]: What's the quantity that needs to fit in the one spare byte for 32-bit arithmetic? Is it still guaranteed to fit in one byte for 64-bit arithmetic?

Comment (maxtaco) [inline openpgp/packet/public_key_v3.go:103]: I believe it's to make sure there's no sign bit confusion

**Comment by maxtaco** [comment]
My general strategy was just to change the type of `E` to `int64` and then fix the compile errors as they came up. The actually use of the int is fine, since it's thrown up into a `big.Int`

**Comment by oconnor663** [comment]
No other comments, other than my general worry about changing crypto code. What's the class of keys that require this big E?

**Comment by maxtaco** [comment]
I've seen 3 offenders to far.  It's not changing the code that has me worried, as much as having to fork and missing security updates in the future.

**Pull request merged**